### ISSUE TITLE
index out of range [1] with length 1

### ISSUE THREAD

**Issue by imsodin** [maruel/panicparse#56]
When calling `pp` with the trace linked below, I get the following result:

```
panic: deadlock detected at fmut

panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
github.com/maruel/panicparse/v2/stack.(*Stack).less(0xc00052d2d0, 0xc00052d480, 0xc000382300)
	/media/ext4_data/Coding/go/pkg/mod/github.com/maruel/panicparse/v2@v2.1.0/stack/stack.go:525 +0x5ef
github.com/maruel/panicparse/v2/stack.(*Signature).less(0xc00052d440, 0xc00052d290, 0x7f15c1c85101)
	/media/ext4_data/Coding/go/pkg/mod/github.com/maruel/panicparse/v2@v2.1.0/stack/stack.go:660 +0x72
github.com/maruel/panicparse/v2/stack.(*Snapshot).Aggregate.func1(0x33, 0x32, 0x1)
	/media/ext4_data/Coding/go/pkg/mod/github.com/maruel/panicparse/v2@v2.1.0/stack/bucket.go:86 +0x87
sort.insertionSort_func(0xc0000d9b00, 0xc00000e7a0, 0x28, 0x3c)
	/usr/lib/go/src/sort/zfuncversion.go:12 +0xb4
sort.stable_func(0xc0000d9b00, 0xc00000e7a0, 0xcc)
	/usr/lib/go/src/sort/zfuncversion.go:167 +0x51
sort.SliceStable(0x5c55a0, 0xc00000e780, 0xc0000d9b00)
	/usr/lib/go/src/sort/slice.go:27 +0xcd
github.com/maruel/panicparse/v2/stack.(*Snapshot).Aggregate(0xc00010d810, 0x2, 0xc0000d9ca8)
	/media/ext4_data/Coding/go/pkg/mod/github.com/maruel/panicparse/v2@v2.1.0/stack/bucket.go:80 +0x7a5
github.com/maruel/panicparse/v2/internal.processInner(0x64e000, 0xc0000b2008, 0xc0000dcf00, 0x2, 0x2, 0x0, 0x0, 0x0, 0x0, 0xc00010d810, ...)
	/media/ext4_data/Coding/go/pkg/mod/github.com/maruel/panicparse/v2@v2.1.0/internal/main.go:133 +0x212
github.com/maruel/panicparse/v2/internal.process(0x64dfe0, 0xc0000b2000, 0x64e000, 0xc0000b2008, 0xc0000dcf00, 0x2, 0x2, 0xc000090101, 0x0, 0x0, ...)
	/media/ext4_data/Coding/go/pkg/mod/github.com/maruel/panicparse/v2@v2.1.0/internal/main.go:162 +0x3b1
github.com/maruel/panicparse/v2/internal.Main(0x0, 0x0)
	/media/ext4_data/Coding/go/pkg/mod/github.com/maruel/panicparse/v2@v2.1.0/internal/main.go:314 +0x7f3
main.main()
	/media/ext4_data/Coding/go/pkg/mod/github.com/maruel/panicparse/v2@v2.1.0/cmd/pp/main.go:19 +0x26
```

trace-file: https://gist.github.com/imsodin/ece52ebaf298477e314fa1f0cbaf5f85

I get the same panic with v2.0.2, and v1.5.0 works.

**Comment by maruel** [comment]
Thanks for the report! I reproduced.

**Comment by maruel** [comment]
I added a recover() in stack.Snapshot.Aggregate(), printed the goroutine ids and identified the goroutines causing this: 11, 12, 52, 61, 52, 65. This permitted me creating a smaller repro. I used the triple 11, 52, 55.

```
panic: deadlock detected at fmut

goroutine 11 [select, 55 minutes]:
github.com/thejerf/suture.(*Supervisor).Serve(0x128c2000)
  github.com/thejerf/suture@v4.0.0+incompatible/supervisor.go:467 +0x2b8
github.com/thejerf/suture.(*Supervisor).runService.func1(0x128d5220, 0x3, 0x32ab0ae8, 0x12856000)
  github.com/thejerf/suture@v4.0.0+incompatible/supervisor.go:643 +0x69
created by github.com/thejerf/suture.(*Supervisor).runService
  github.com/thejerf/suture@v4.0.0+incompatible/supervisor.go:631 +0x4b

goroutine 52 [select, 55 minutes]:
github.com/thejerf/suture.(*Supervisor).Serve(0x128d4460)
  github.com/thejerf/suture@v4.0.0+incompatible/supervisor.go:467 +0x2b8
created by github.com/thejerf/suture.(*Supervisor).ServeBackground
  github.com/thejerf/suture@v4.0.0+incompatible/supervisor.go:420 +0x33

goroutine 55 [select, 55 minutes]:
github.com/thejerf/suture.(*Supervisor).Serve(0x128d4000)
  github.com/thejerf/suture@v4.0.0+incompatible/supervisor.go:467 +0x2b8
github.com/thejerf/suture.(*Supervisor).runService.func1(0x128d4460, 0x1, 0x1073b60, 0x12852380)
  github.com/thejerf/suture@v4.0.0+incompatible/supervisor.go:643 +0x69
created by github.com/thejerf/suture.(*Supervisor).runService
  github.com/thejerf/suture@v4.0.0+incompatible/supervisor.go:631 +0x4b
```

I created a new unit test that causes this crash. Now time to investigate why and how to fix.

**Comment by maruel** [comment]
Published v1.6.1 and v2.1.1 to address this issue. Primed go proxy so they will be fetched right away.
